adagrams/game.py

```diff
@@ -1,11 +1,108 @@
+import random
+
+    
 def draw_letters():
-    pass
+    letter_pool = {
+        'A': 9, 
+        'B': 2, 
+        'C': 2, 
+        'D': 4, 
+        'E': 12, 
+        'F': 2, 
+        'G': 3, 
+        'H': 2, 
+        'I': 9, 
+        'J': 1, 
+        'K': 1, 
+        'L': 4, 
+        'M': 2, 
+        'N': 6, 
+        'O': 8, 
+        'P': 2, 
+        'Q': 1, 
+        'R': 6, 
+        'S': 4, 
+        'T': 6, 
+        'U': 4, 
+        'V': 2, 
+        'W': 2, 
+        'X': 1, 
+        'Y': 2, 
+        'Z': 1
+    }
+    # create list of strings 'letters'
+    letters = []
+    # create empty list 'hand'
+    hand = []
+    # build a random hand of 10 letters for the user
+    while len(hand) < 10:
+        letter_pool_keys = letter_pool.keys()
+        letter_list = list(letter_pool_keys)
+        if len(letter_list) > 0:
+            letter = random.choice(letter_list)
```

Comment:
So, you handled the constraints of the weights correctly (making sure that you don't have too many instances of one letter in the list). However, with `letter = random.choice(letter_list)` all the letters have an equal chance of being chosen, whereas we want certain letters to be have a greater chance (`E` having a value of `12` should be 12x as likely to be chosen than `Z` being that it has only value of `1`. We can do that with an implementation like this: 
```python
letters = []
for letter in LETTER_POOL:
    letters += letter * LETTER_POOL[letter]
```
This will create a list of letters where each `letter` will be added to the list however many times that correlates with its value pair. ` 'S' * 4 => ['S', 'S', 'S', 'S']`